### PR TITLE
Fix several issues with the Google MQTT demo

### PR DESCRIPTION
**Pull request by d3zd3z**
A change to the net configuration has broken compilation. In addition, the instructions are difficult to follow, and somewhat misleading.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]
LGTM

File: samples/net/cloud/google_iot_mqtt/prj.conf [hunk 1]
```diff
@@ -6,6 +6,8 @@ CONFIG_NET_TCP=y
 CONFIG_NET_DHCPV4=y
 CONFIG_STDOUT_CONSOLE=y
 
+CONFIG_LEGACY_TIMEOUT_API=y
```
Comment (jukkar) [inline samples/net/cloud/google_iot_mqtt/prj.conf:9]: I do not think we should re-enable the legacy timeout as we are trying to get rid of it. I created a separate PR #25122 that converts the google iot sample to new timeouts so we do not need to apply this specific commit. The rest of the PR looks fine.

**Review by jukkar — CHANGES_REQUESTED** [review]

**Review by pfalcon — APPROVED** [review]

File: samples/net/cloud/google_iot_mqtt/prj.conf [hunk 2]
```diff
@@ -57,3 +57,9 @@ CONFIG_MBEDTLS_ENABLE_HEAP=y
 CONFIG_MBEDTLS_HEAP_SIZE=56240
 CONFIG_MBEDTLS_USER_CONFIG_ENABLE=y
 CONFIG_MBEDTLS_USER_CONFIG_FILE="user-tls.conf"
+
+CONFIG_CLOUD_CLIENT_ID="projects/<PROJECT_ID>/locations/<REGION>/registries/<REGISTRY_ID>/devices/"
```
Comment (pfalcon) [inline samples/net/cloud/google_iot_mqtt/prj.conf:63]: Ideally, these should be preceding by comment saying something like: "# Fill in values based on your Google IoT Cloud configuration, see README.rst for more details".

**Review by dieselburner — CHANGES_REQUESTED** [review]

File: samples/net/cloud/google_iot_mqtt/prj.conf (hunk 2 above)
Comment (d3zd3z) [reply samples/net/cloud/google_iot_mqtt/prj.conf:63]: Do you know if comments actually work in Kconfig files? I know that Kconfig parses comments as `# CONFIG_FOO is not set`. It looks safe and we use it, but I wanted to make sure.

**Comment by d3zd3z** [comment]
Removed the legacy timer API patch, as it is no longer needed, and applied suggestions in the comments.

**Review by jukkar — APPROVED** [review]

**Comment by dieselburner** [comment]
Thanks, I got it working because of your changes.

**Pull request merged**